lib/ansible/modules/cloud/azure/azure_rm_servicebus.py

```diff
@@ -0,0 +1,587 @@
+        required: true
+    name:
+        description:
+            - name of the route.
+        required: true
+    state:
+        description:
+            - Assert the state of the route. Use 'present' to create or update and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+    namespace:
+        description:
+            - Servicebus namespace name.
+            - A namespace is a scoping container for all messaging components.
+            - Multiple queues and topics can reside within a single namespace, and namespaces often serve as application containers.
+            - Required when C(type) is not C(namespace).
+    location:
+        description:
+            - Namespace location.
+    type:
+        description:
+            - Type of the messaging application.
+            - Queue enables you to store messages until the receiving application is available to receive and process them.
+            - Topic and subscriptions enable 1:n relationships between publishers and subscribers.
+        choices:
+            - namespace
+            - queue
+            - topic
+            - subscription
+    subscription_topic_name:
+        description:
+            - Topic name which the subscription subscribe to.
+            - Required when C(type) is set to C(subscription).
+    auto_delete_on_idle_in_seconds:
+        description:
+            - Time idle interval after which a queue is automatically deleted.
+            - The minimum duration is 5 minutes.
+        type: int
+    dead_lettering_on_message_expiration:
+        description:
+            - A value that indicates whether a queue has dead letter support when a message expires.
+        type: bool
+    dead_lettering_on_filter_evaluation_exceptions:
+        description:
+            - Value that indicates whether a subscription has dead letter support on filter evaluation exceptions.
+        type: bool
+    default_message_time_to_live_seconds:
+        description:
+            - Default message timespan to live value.
+            - This is the duration after which the message expires, starting from when the message is sent to Service Bus.
+            - This is the default value used when TimeToLive is not set on a message itself.
+        type: int
+    enable_batched_operations:
+        description:
+            - Value that indicates whether server-side batched operations are enabled.
+        type: bool
+    enable_express:
+        description:
+            - Value that indicates whether Express Entities are enabled.
+            - An express topic or queue holds a message in memory temporarily before writing it to persistent storage.
+        type: bool
+    enable_partitioning:
+        description:
+            - A value that indicates whether the topic or queue is to be partitioned across multiple message brokers.
+        type: bool
+    forward_dead_lettered_messages_to:
```

Comment:
it's hard to say whether this parameter applies to namespace / queue / topic / subscription?
this applies to most of the parameters that are not required_if.

Reply:
actually, most of the parameters are required for all of queue/topic/subscriptions